Orange/widgets/evaluate/owparameterfitter.py

```diff
@@ -318,93 +368,150 @@ def _add_plot(self):
+
+    def __on_min_max_changed(self):
+        self.type = self.FROM_RANGE
+        self._settings_changed()
 
-    def __on_setting_changed(self):
+    def __on_manual_changed(self):
+        self.type = self.MANUAL
+        self._settings_changed()
+
+    def _settings_changed(self):
         self._update_preview()
         self.commit.deferred()
 
     @property
     def fitted_parameters(self) -> list:
-        if not self._learner or not self._data:
+        if not self._learner:
             return []
-        return self._learner.fitted_parameters(self._data) \
-            if isinstance(self._learner, Fitter) \
-            else self._learner.fitted_parameters()
+        if isinstance(self._learner, Fitter):
+            if not self._data:
+                return []
+            return self._learner.fitted_parameters(self._data)
+        else:
+            return self._learner.fitted_parameters()
 
     @property
     def initial_parameters(self) -> dict:
-        if not self._learner or not self._data:
+        if not self._learner:
             return {}
-        return self._learner.get_params(self._data) \
-            if isinstance(self._learner, Fitter) else self._learner.params
+        if isinstance(self._learner, Fitter):
+            if not self._data:
+                return {}
+            return self._learner.get_params(self._data)
+        return self._learner.params
 
     @property
-    def steps(self) -> Iterable[int]:
+    def steps(self) -> tuple[int]:
+        self.Error.min_max_error.clear()
+        self.Error.manual_steps_error.clear()
         if self.type == self.FROM_RANGE:
-            step = 1
-            diff = self.maximum - self.minimum
-            if diff > 0:
-                exp = int(np.ceil(np.log10(diff + 1))) - 1
-                step = int(10 ** exp)
-            return range(self.minimum, self.maximum + step, step)
+            return self._steps_from_range()
         else:
-            try:
-                return [int(s) for s in self.manual_steps.split(",")]
-            except ValueError:
-                return []
+            return self._steps_from_manual()
+
+    def _steps_from_range(self) -> tuple[int]:
+        if self.maximum < self.minimum:
+            self.Error.min_max_error()
+            return ()
+
+        if self.minimum == self.maximum:
+            return (self.minimum, )
+
+        diff = self.maximum - self.minimum
+        # This should give between 10 and 15 steps
+        exp = max(0, int(np.ceil(np.log10(diff / 1.5))) - 1)
+        step = int(10 ** exp)
+        return (self.minimum,
+                *range((self.minimum // step + 1) * step, self.maximum, step),
+                self.maximum)
+
+    def _steps_from_manual(self) -> tuple[int]:
+        steps = self.manual_steps
+        if not steps:
+            return ()
+        try:
+            steps = tuple(sorted(map(int, steps.split(","))))
+            self.manual_steps = ", ".join(map(str, steps))
```

Comment:
We can discuss this. I don't see a reason, though, why anybody would want unsorted numbers. It is more likely that one will want to insert a number, so this feature is more useful than annoying.

Reply:
Fine. 

But even more useful I find checking that `manual_steps` are within `fitted_parameter.min` and `fitted_parameter.max`.

Reply:
I forgot about this. But now that you mentioned: what about letting the user go out of range, and just show a warning? Just an idea.

Reply:
That's ok with me, as long as showing the warning also implies to not fitting the model.